xonsh/built_ins.py

```diff
@@ -637,15 +637,41 @@ def run_subproc(cmds, captured=True):
             hist.last_cmd_out = output
     if not prev_is_proxy and hist.last_cmd_rtn > 0 and ENV.get('RAISE_SUBPROC_ERROR'):
         raise CalledProcessError(hist.last_cmd_rtn, aliased_cmd, output=output)
-    if captured:
+    if captured=='stdout':
         return output
+    elif captured=='object':
+        return CompletedProcess(prev_proc, output)
```

Comment:
When using subclassed `subprocess.CompletedProcess`, its original constructor can be used here.

So this line would become:

``` python
return CompletedProcess(prev_proc.args, prev_proc.returncode, output, erroroutput)
```

`erroroutput` needs to be read from `stderr` before this call, similarly as `output` is read from `stdout`.